### PR TITLE
AP_Motors: Added DJI_X motor setup for hexacopter and octacopter

### LINKED ISSUE #13395: Copter: add DJI_X motor setup for hexacopter and octacopter
This request comes from [this discussion in the Copter-4.0 forums](https://discuss.ardupilot.org/t/copter-4-0-1-released/51505/29).

We should add DJI_X motor setups for hexacopter and octacopters.  currently this frame type is only available for quadcopters.

DJI_X always has the motor numbers increasing counter clockwise and the front right motor is always 1.

### PR DESCRIPTION
**Pull request by rishabsingh3003**
This PR fixes #13395.
As suggested by @rmackay9 I took the motor numbers increasing counter-clockwise and the front right motor as 1. 
Please let me know if any other changes are required.

### CODE REVIEW

**Comment by auturgy** [comment]
A few things: simplest thing for you to do would be to have a look at this PR: https://github.com/ArduPilot/ardupilot/pull/13402
which has similar changes plus the required changes to sitl and the class header.
With my branch, yaw isn't working in SITL so I wasn't going to push it til I figured that out - the mixer is the same as here though.  Feel free to cherry pick etc, as I'm not sure when I'll get back to it.

**Comment by rishabsingh3003** [comment]
@auturgy Thanks! I had a look at your PR and added SITL support for the new motor mixes. I tested in SITL and it works well. Just set the correct FRAME_CLASS and FRAME_TYPE param.

File: libraries/AP_HAL_SITL/SITL_cmdline.cpp [hunk 1]
```diff
@@ -112,7 +112,9 @@ static const struct {
     { "djix",               MultiCopter::create },
     { "cwx",                MultiCopter::create },
     { "hexa",               MultiCopter::create },
+    { "djihexa",            MultiCopter::create },
```
Comment (rmackay9) [inline libraries/AP_HAL_SITL/SITL_cmdline.cpp]: Looking pretty good.  I'd prefer if we kept the frame name at the front and then add the "dji" part.  so let's make it "hexa-dji" and "octa-dji".  I see that some of the other existing frame names like "bfxrev" and "djix" are missing the quad so we should fix that as well but that doesn't need to be part of this PR, in fact, it's probably best that it's not.

Comment (rmackay9) [reply libraries/AP_HAL_SITL/SITL_cmdline.cpp]: If you can update this then I'll merge it, thanks!!

**Comment by rishabsingh3003** [comment]
@rmackay9, I have fixed this issue! Thanks :) I will push another PR for the secondary issue you have mentioned.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**